fuel-core/src/schema/coin.rs

```
@@ -168,4 +178,46 @@ impl CoinQuery {
         )
         .await
     }
+
+    async fn coins_to_spend(
+        &self,
+        ctx: &Context<'_>,
+        #[graphql(desc = "The Address of the utxo owner")] owner: HexString256,
+        #[graphql(desc = "The total amount of each asset type to spend")] spend_query: Vec<
+            SpendQueryElementInput,
+        >,
+        #[graphql(desc = "The max number of utxos that can be used")] max_inputs: Option<i32>,
+    ) -> async_graphql::Result<Vec<Coin>> {
+        let owner: Address = owner.0.into();
+        let spend_query: Vec<(FuelTxColor, u64)> = spend_query
+            .iter()
+            .map(|e| (FuelTxColor::from(e.color.0), e.amount.0))
+            .collect();
+        let max_inputs: u8 = max_inputs
+            .map(u8::try_from)
+            .map(Result::unwrap)
```

why not return an error?